pkg/cloud/azure/services/virtualmachines/virtualmachines.go

```diff
@@ -413,3 +444,82 @@ func generateImagePlan(image machinev1.Image) *compute.Plan {
 		Product:   to.StringPtr(image.Offer),
 	}
 }
+
+func generateDataDisks(vmSpec *Spec) ([]compute.DataDisk, error) {
+	dataDiskLuns := make(map[int32]struct{})
+	dataDiskNames := make(map[string]struct{})
+	reg := regexp.MustCompile(`^[a-zA-Z0-9](?:[\w\.-]*[a-zA-Z0-9])?$`)
```

Comment:
Please add a comment to explain what this regex achieves so that future us doesn't have to study the regex